multimodal/src/autogluon/multimodal/learners/default_learner.py

```diff
@@ -0,0 +1,2906 @@
+        #     )
+        #     softmax_regression_weight = OmegaConf.select(
+        #         self._config, "distiller.softmax_regression_weight", default=0.0
+        #     )
+        #     use_raw_features = OmegaConf.select(self._config, "distiller.use_raw_features", default=False)
+        #     task = DistillerLitModule(
+        #         student_model=model,
+        #         teacher_model=teacher_model,
+        #         matches=config.distiller.matches,
+        #         critics=critics,
+        #         baseline_funcs=baseline_funcs,
+        #         hard_label_weight=config.distiller.hard_label_weight,
+        #         soft_label_weight=config.distiller.soft_label_weight,
+        #         softmax_regression_weight=softmax_regression_weight,
+        #         temperature=config.distiller.temperature,
+        #         output_feature_loss_weight=output_feature_loss_weight,
+        #         hard_label_loss_func=loss_func,
+        #         soft_label_loss_func=soft_label_loss_func,
+        #         softmax_regression_loss_func=softmax_regression_loss_func,
+        #         output_feature_adaptor=output_feature_adaptor,
+        #         output_feature_loss_func=output_feature_loss_func,
+        #         rkd_loss_func=rkd_loss_func,
+        #         **metrics_kwargs,
+        #         **optimization_kwargs,
+        #     )
+        # elif self._problem_type == NER:
+        #     task = NerLitModule(
+        #         model=model,
+        #         loss_func=loss_func,
+        #         efficient_finetune=OmegaConf.select(config, "optimization.efficient_finetune"),
+        #         mixup_fn=mixup_fn,
+        #         mixup_off_epoch=OmegaConf.select(config, "data.mixup.turn_off_epoch"),
+        #         model_postprocess_fn=model_postprocess_fn,
+        #         trainable_param_names=trainable_param_names,
+        #         **metrics_kwargs,
+        #         **optimization_kwargs,
+        #     )
+        # elif self._problem_type == OBJECT_DETECTION:
+        #     task = MMDetLitModule(
+        #         model=model,
+        #         **metrics_kwargs,
+        #         **optimization_kwargs,
+        #     )
+        # else:
+        #     task = LitModule(
+        #         model=model,
+        #         loss_func=loss_func,
+        #         efficient_finetune=OmegaConf.select(config, "optimization.efficient_finetune"),
+        #         mixup_fn=mixup_fn,
+        #         mixup_off_epoch=OmegaConf.select(config, "data.mixup.turn_off_epoch"),
+        #         model_postprocess_fn=model_postprocess_fn,
+        #         trainable_param_names=trainable_param_names,
+        #         skip_final_val=OmegaConf.select(config, "optimization.skip_final_val", default=False),
+        #         **metrics_kwargs,
+        #         **optimization_kwargs,
+        #     )
+
+        task = self._setup_task_lightning_module(
```

Comment:
Consider using `_get_lightning_module()` to be more clear.

Reply:
Done